### ISSUE TITLE
Goodput Ratio Calculation Accuracy ?

### ISSUE THREAD

**Issue by alihushyar** [ntop/nDPI#1853]
## Describe the bug
Questioning how goodput ratio is calculated when pcap includes retransmissions.  

### Expected behavior
I would expect goodput is simply the cumulative non-zero payload packet size sent from A->B not counting retransmissions and duplicate packets.   
### Obtained behavior
Attached is a sample pcap showing how the client is retransmitting every packet.  My goodput calculation is different then what ndpIReader shows.  ndpiReader ratio calc is 65 out and 94 in.  My calc is 33 out and 94 in.  

## nDPI Environment (please complete the following information):
* OS name: Ubuntu.
* OS version: 20.04
* Architecture: [amd
* nDPI version: nDPI 4.5.0-3751-2c3f56d4

``` shell
ndpiReader -v 2 -i  ~/projects/agentslim/pcaps/CF_TLS_21_2.pcap

[CF_TLS_21_2.zip](https://github.com/ntop/nDPI/files/10422479/CF_TLS_21_2.zip)

**Comment by IvanNardi** [comment]
@alihushyar, could you properly attached the pcap, please?

**Comment by alihushyar** [comment]
[CF_TLS_21_2.zip](https://github.com/ntop/nDPI/files/10428308/CF_TLS_21_2.zip)

**Comment by IvanNardi** [comment]
While it seems you are right, fixing the code is not trivial.
The core reason is that  the code checking if a (TCP) packet is a retransmission or not, is executed from the nDPI core library itself (via `ndpi_detection_process_packet()` -> `ndpi_connection_tracking()`): this functions are only called for the very first packets of the flow, only until all the DPI work has been done.  After that point we don't know anymore if a packet is a duplicate or not.

Bottom line: ignoring retransmissions while processing the flow is easy; ignoring them after the flow has completely been processed, it is not.
